File: antlr-python/ChatErrorListener.py
import sys
from antlr4 import *
from ChatParser import ChatParser
from ChatListener import ChatListener
from antlr4.error.ErrorListener import *
import io

class ChatErrorListener(ErrorListener):

    def __init__(self, output):
        self.output = output        
        self._symbol = ''
    
    def syntaxError(self, recognizer, offendingSymbol, line, column, msg, e):        
        self.output.write(msg)
        if offendingSymbol is not None:
            self._symbol = offendingSymbol.text
        else:            
            self._symbol = recognizer.getTokenErrorDisplay(offendingSymbol);        

    @property        
    def symbol(self):
        return self._symbol

File: antlr-js/ChatErrorListener.js
import antlr4 from 'antlr4';
import ChatLexer from './ChatLexer.js';
import ChatParser from './ChatParser.js';
import ChatListener  from './ChatListener.js';

export default class ChatErrorErrorListener extends antlr4.error.ErrorListener {
    constructor(res) {
        super();
        this.Res = res;
        this.symbol = '';
    }

    syntaxError(recognizer, offendingSymbol, line, column, msg, e) {            
        this.Res.write(msg);
        
        if(offendingSymbol == null)
            this.symbol = recognizer.getTokenErrorDisplay(offendingSymbol);
        else
            this.symbol = offendingSymbol.text;
    }
}